src/org.hdfgroup.hdfview/hdf/view/TableView/DataProviderFactory.java

```diff
@@ -1672,6 +1680,79 @@ public Object getDataValue(Object obj, int index)
+            if (isFLT16)
+                theValue = Short.toString(Float.floatToFloat16(Float.parseFloat((String)newValue)));
+
+            super.setDataValue(columnIndex, rowIndex, theValue);
+        }
+
+        /**
+         * When a CompoundDataProvider wants to pass a List of data down to a nested CompoundDataProvider, or
+         * when a top-level container DataProvider (such as an ArrayDataProvider) wants to hand data down to a
+         * base CompoundDataProvider, we need to pass down a List of data and the new value, plus a field and
+         * row index. This method is for facilitating this behavior.
+         *
+         * In general, all "container" DataProviders that have a "container" base DataProvider should call
+         * down into their base DataProvider(s) using this{}, method, in order to ensure that buried
+         * CompoundDataProviders get handled correctly. When their base DataProvider is not a "container"
+         * type, the method setDataValue(index, Object, Object) should be used instead.
+         *
+         * For atomic type DataProviders, we treat this method as directly calling into setDataValue(index,
+         * Object, Object) using the passed rowIndex. However, this method should, in general, not be called
+         * by atomic type DataProviders.
+         *
+         * @param columnIndex the column
+         * @param rowIndex    the row
+         * @param bufObject   the data object
+         * @param newValue    the new data object
+         */
+        public void setDataValue(int columnIndex, int rowIndex, Object bufObject, Object newValue)
+        {
+            Object newbufObject = bufObject;
+            if (isFLT16)
+                newbufObject = Float.float16ToFloat((short)bufObject);
+            setDataValue(rowIndex, newbufObject, newValue);
+        }
+
+        /**
+         * When a parent HDFDataProvider (such as an ArrayDataProvider) wants to set a data value by routing
+         * the operation through its base HDFDataProvider, the parent HDFDataProvider will generally know the
+         * direct index to have the base provider use. This method is to facilitate this kind of behavior.
+         *
+         * Note that this method takes two Object parameters, one which is the object that the method should
+         * set its data inside of and one which is the new value to set. This is to be able to nicely support
+         * nested compound DataProviders.
+         *
+         * @param index     the index into the data array
+         * @param bufObject the data object
+         * @param newValue  the new data object
+         */
+        public void setDataValue(int index, Object bufObject, Object newValue)
+        {
+            Object newbufObject = bufObject;
+            if (isFLT16)
+                newbufObject = Float.float16ToFloat((short)bufObject);
+            try {
+                setDataValue(index, newbufObject, newValue);
```

Comment:
I think this `setDataValue` call is just recursively calling itself, which will lead to a stack overflow when editing float16 values inside an array type. I do believe we need this `setDataValue` variant though to support array types.

Reply:
Yep, exactly what happened.